datasafe-simple-adapter/datasafe-simple-adapter-impl/src/main/java/de/adorsys/datasafe/simple/adapter/impl/SimpleDatasafeServiceImpl.java

```diff
@@ -226,25 +226,30 @@ private static SystemRootAndStorageService useAmazonS3(AmazonS3DFSCredentials df
 
         log.info(lsf.toString());
         AmazonS3ClientBuilder amazonS3ClientBuilder = AmazonS3ClientBuilder.standard()
-                .withCredentials(
-                        new AWSStaticCredentialsProvider(
-                                new BasicAWSCredentials(
-                                        amazonS3DFSCredentials.getAccessKey(),
-                                        amazonS3DFSCredentials.getSecretKey()))
-                )
-                .enablePathStyleAccess();
+            .withCredentials(
+                new AWSStaticCredentialsProvider(
+                    new BasicAWSCredentials(
+                        amazonS3DFSCredentials.getAccessKey(),
+                        amazonS3DFSCredentials.getSecretKey()))
+            );
 
-        boolean useEndpoint = !amazonS3DFSCredentials.getUrl().equals(AMAZON_URL)
-                && !amazonS3DFSCredentials.getUrl().startsWith(S3_PREFIX);
+        boolean useEndpoint = !amazonS3DFSCredentials.getUrl().matches(AMAZON_URL)
+            && !amazonS3DFSCredentials.getUrl().startsWith(S3_PREFIX);
+        lsf = new LogStringFrame();
         if (useEndpoint) {
+            lsf.add("not real amazon, so use pathStyleAccess");
             AwsClientBuilder.EndpointConfiguration endpoint = new AwsClientBuilder.EndpointConfiguration(
-                    amazonS3DFSCredentials.getUrl(),
-                    amazonS3DFSCredentials.getRegion()
+                amazonS3DFSCredentials.getUrl(),
+                amazonS3DFSCredentials.getRegion()
             );
-            amazonS3ClientBuilder.withEndpointConfiguration(endpoint);
+            amazonS3ClientBuilder
+                .withEndpointConfiguration(endpoint)
+                .enablePathStyleAccess();
         } else {
+            lsf.add("real amazon, so use bucketStyleAccess");
             amazonS3ClientBuilder.withRegion(amazonS3DFSCredentials.getRegion());
         }
+        log.info(lsf.toString());
```

Comment:
Please use `log.info("{}", lsf);` for logging. This will avoid situations when lsf.toString() yields value `data {} to write` that will cause an issue (while improbable it is very easy to avoid)

Reply:
done